src/prommis/hydrogen_decrepitation/hydrogen_decrepitation_furnace.py

```diff
@@ -0,0 +1,1651 @@
+        )
+        self.mw_comp_product["Nd"] = self.am_Nd
+        self.mw_comp_product["Nd2Fe14B"] = self.mw_Nd2Fe14B
+
+        # standard enthalpy of formation compound at 298.15 K
+        self.enth0_Nd = Param(initialize=326.9e3, units=pyunits.J / pyunits.mol)  # [2]
+        self.enth0_Nd2Fe14B = Param(
+            initialize=-93.39e3, units=pyunits.J / pyunits.mol
+        )  # [4]
+
+        # heat capacity modeled as linear function of temperature T in K.
+        # Cp = Cp0 + Cp1*T
+        self.cp0_Nd = Param(
+            initialize=22.44, units=pyunits.J / pyunits.mol / pyunits.K
+        )  # [3]
+        self.cp0_Nd2Fe14B = Param(
+            initialize=272.59, units=pyunits.J / pyunits.mol / pyunits.K
+        )
+
+        self.cp1_Nd = Param(
+            initialize=0.0162, units=pyunits.J / pyunits.mol / pyunits.K**2
+        )
+        self.cp1_Nd2Fe14B = Param(
+            initialize=0.6487, units=pyunits.J / pyunits.mol / pyunits.K**2
+        )
+
+        # standard heat of formation of impurity species
+        self.enth0_comp_impurity = Param(
+            self.impurity_list, mutable=True, units=pyunits.J / pyunits.mol
+        )
+        self.enth0_comp_impurity["Nd"] = self.enth0_Nd
+        self.enth0_comp_impurity["Nd2Fe14B"] = self.enth0_Nd2Fe14B
+
+        # standard heat of formation of solid product species
+        self.enth0_comp_product = Param(
+            self.product_list, mutable=True, units=pyunits.J / pyunits.mol
+        )
+        self.enth0_comp_product["Nd"] = self.enth0_Nd
+        self.enth0_comp_product["Nd2Fe14B"] = self.enth0_Nd2Fe14B
+
+        # 1st heat capacity coefficient of impurity species
+        self.cp0_comp_impurity = Param(
+            self.impurity_list, mutable=True, units=pyunits.J / pyunits.mol / pyunits.K
+        )
+        self.cp0_comp_impurity["Nd"] = self.cp0_Nd
+        self.cp0_comp_impurity["Nd2Fe14B"] = self.cp0_Nd2Fe14B
+
+        # 1st heat capacity coefficient of solid product species
+        self.cp0_comp_product = Param(
+            self.product_list, mutable=True, units=pyunits.J / pyunits.mol / pyunits.K
+        )
+        self.cp0_comp_product["Nd"] = self.cp0_Nd
+        self.cp0_comp_product["Nd2Fe14B"] = self.cp0_Nd2Fe14B
+
+        # 2nd heat capacity coefficient of impurity species
+        self.cp1_comp_impurity = Param(
+            self.impurity_list,
+            mutable=True,
+            units=pyunits.J / pyunits.mol / pyunits.K**2,
+        )
+        self.cp1_comp_impurity["Nd"] = self.cp1_Nd
+        self.cp1_comp_impurity["Nd2Fe14B"] = self.cp1_Nd2Fe14B
+
+        # 2nd heat capacity coefficient of solid product species
+        self.cp1_comp_product = Param(
+            self.product_list,
+            mutable=True,
+            units=pyunits.J / pyunits.mol / pyunits.K**2,
+        )
+        self.cp1_comp_product["Nd"] = self.cp1_Nd
+        self.cp1_comp_product["Nd2Fe14B"] = self.cp1_Nd2Fe14B
```

Comment:
Why are we making scalar parameters, then creating indexed parameters, then initializing those indexed parameters with the values of the scalar parameters?